modules/web-console/frontend/app/components/page-queries/components/queries-notebook/controller.ts

```diff
@@ -963,19 +963,24 @@ export class NotebookCtrl {
             const cluster$ = agentMgr.connectionSbj.pipe(
                 pluck('cluster'),
                 distinctUntilChanged(),
-                tap((cluster) => {
-                    this.clusterIsAvailable = (!!cluster && cluster.active === true) || agentMgr.isDemoMode();
+            );
+
+            const checkState$ = combineLatest(
+                cluster$,
+                agentMgr.currentCluster$.pipe(pluck('state'), distinctUntilChanged())
+            ).pipe(
+                tap(([cluster, state]) => {
+                    if (state === 'CONNECTED')
+                        this.clusterIsAvailable = (!!cluster && cluster.active === true) || agentMgr.isDemoMode();
                 })
             );
 
-            this.refresh$ = cluster$.pipe(
-                switchMap((cluster) => {
-                    if (!cluster && !agentMgr.isDemoMode()) {
-                        return of(EMPTY).pipe(
-                            tap(() => {
-                                $scope.caches = [];
-                            })
-                        );
+            this.refresh$ = checkState$.pipe(
+                switchMap(([cluster, state]) => {
+                    if (state !== 'CONNECTED' || (!cluster && !agentMgr.isDemoMode())) {
```

Comment:
I'd move side-effects into tap before `switchMap`.

Reply:
Side effect moved to tap function.